### ISSUE TITLE
AddImport produces corrupt binary on linux

### ISSUE THREAD

**Issue by superewald** [secana/PeNet#245]
Adding an import entry to a PeFile produces corrupt output when running on linux.

## details

I'm using this library to patch an applications IAT and load a custom dll. When running on windows it works fine but running on linux the output binary is not executed properly anymore.

## reproduce

```
var peFile = new PeFile(@"some/application.exe");
peFile.AddImport("custom.dll", "function");

File.WriteAllBytes(@"some/application.exe", peFile.RawFile.ToArray());
```

## logs

Heres the output of running the patched application on linux (using wine, running on windows has same result):

```
$ wine BLR.exe 
wine: Call from 0x7bc7c9e1 to unimplemented function ntoskrnl.exe.RtlPcToFileHeader, aborting
wine: Unimplemented function ntoskrnl.exe.RtlPcToFileHeader called at address 000000007BC7C9E1 (thread 0016), starting debugger...
000f:err:service:process_send_command service protocol error - failed to read pipe r = 0  count = 0!
0036:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\NDIS": 00000001
0009:err:module:LdrInitializeThunk "Proxy.dll" failed to initialize, aborting
0009:err:module:LdrInitializeThunk Initializing dlls for L"Z:\\media\\blacklightretribution\\Binaries\\Win32\\BLR.exe" failed, status c0000005
```

**Comment by secana** [comment]
@superewald That is some very strange behavior. I'll try to have look in the next days, but a.t.m. I cannot think of any reason there should be a difference between Windows and Linux.

**Comment by superewald** [comment]
Yes, I was very confused too. I've noticed the patched binary on linux is also ~3.6 MB larger than the patched binary on windows.

One thing that confuses me still (but probably isn't related to this issue) is the fact that if I compare the (working) patched and original files, there is just no difference at all (even though the DLL is added to the IAT). I guess I'm missing something about how PE headers work.

I've uploaded 3 files in case you are unable to reproduce:
- [Original binary](https://mega.nz/file/SYU1lJLI#2tZzOSGvLEPfFNcd3T0wYfcq7Om0N0n13xpprgFxF9c)
- [Patched on Windows (working)](https://mega.nz/file/2A8A3BrB#wrDXUOT4fq7EnvPqyosGugMrCx78Goe_feTrh7K-e9U)
- [Patched on Linux](https://mega.nz/file/XEUj2JqZ#q2qXkcrnKxbS2c1MprLZ6gcPrNPl81Vr0Gr8KzhVSvw)

**Comment by secana** [comment]
Thanks for the files. Are you sure the "Patched on Windows" is the correct one?

0f4a732484f566d928c580afdae6ef01c002198dd7158cb6de29b9a4960064c7  Original.exe
0f4a732484f566d928c580afdae6ef01c002198dd7158cb6de29b9a4960064c7  Patched-on-Windows.exe
02cce81e91fe6b98568892f7d921c3568212dd13a0a2b28c33157f1f9efed984  Patched-on-Linux.exe

Original and the patched one have the same hash sum, which means it's the same wile without any changes.
